modules/index-service/src/main/java/org/opencastproject/index/service/impl/IndexServiceImpl.java

```diff
@@ -1740,15 +1740,151 @@ public Source getEventSource(Event event) {
   }
 
   private void updateMediaPackageMetadata(MediaPackage mp, MetadataList metadataList) {
+    String oldSeriesId = mp.getSeries();
     List<EventCatalogUIAdapter> catalogUIAdapters = getEventCatalogUIAdapters();
-    if (catalogUIAdapters.size() > 0) {
+    if (catalogUIAdapters != null) {
       for (EventCatalogUIAdapter catalogUIAdapter : catalogUIAdapters) {
         Opt<MetadataCollection> metadata = metadataList.getMetadataByAdapter(catalogUIAdapter);
         if (metadata.isSome() && metadata.get().isUpdated()) {
           catalogUIAdapter.storeFields(mp, metadata.get());
         }
       }
     }
+
+    // update series catalogs
+    if (!StringUtils.equals(oldSeriesId, mp.getSeries())) {
+      List<String> seriesDcTags = new ArrayList<>();
+      List<String> seriesAclTags = new ArrayList<>();
+      Map<String, List<String>> seriesExtDcTags = new HashMap<>();
+      if (StringUtils.isNotBlank(oldSeriesId)) {
+        // remove series dublincore from the media package
+        for (MediaPackageElement mpe : mp.getElementsByFlavor(MediaPackageElements.SERIES)) {
+          mp.remove(mpe);
+          for (String tag : mpe.getTags()) {
+            seriesDcTags.add(tag);
+          }
+        }
+        // remove series ACL from the media package
+        for (MediaPackageElement mpe : mp.getElementsByFlavor(MediaPackageElements.XACML_POLICY_SERIES)) {
+          mp.remove(mpe);
+          for (String tag : mpe.getTags()) {
+            seriesAclTags.add(tag);
+          }
+        }
+        // remove series extended metadata from the media package
+        try {
+          Opt<Map<String, byte[]>> oldSeriesElementsOpt = seriesService.getSeriesElements(oldSeriesId);
+          for (Map<String, byte[]> oldSeriesElements : oldSeriesElementsOpt) {
+            for (String oldSeriesElementType : oldSeriesElements.keySet()) {
+              for (MediaPackageElement mpe : mp
+                      .getElementsByFlavor(MediaPackageElementFlavor.flavor(oldSeriesElementType, "series"))) {
+                mp.remove(mpe);
+                String elementType = mpe.getFlavor().getType();
+                if (StringUtils.isNotBlank(elementType)) {
+                  for (String tag : mpe.getTags()) {
```

Comment:
If would be easier to understand to put the logic...

```
if (!seriesExtDcTags.containsKey(elementType))
       List<String> tags = new ArrayList<>();
      seriesExtDcTags.put(elementType, tags);
```

... in front of the for loop and then just do:

```
for (String tag : mpe.getTags()) {
    seriesExtDcTags.get(elementType).add(tag);
}
```

Reply:
Ok, will change it.